numpy/_core/tests/test_multiarray.py

```diff
@@ -3345,6 +3345,29 @@ def test_dot(self):
         a.dot(b=b, out=c)
         assert_equal(c, np.dot(a, b))
 
+    @pytest.mark.skipif(IS_WASM, reason="no wasm fp exception support")
+    def test_dot_errstate(self):
```

Comment:
Small nit:  Could you parametrize this to include `half` and/or `longdouble` (just because they currently take the other code paths and can set the warnings).

The point is that double/float should effectively always take the first branch, that way we should cover both error paths.